hwtracer/src/collect/mod.rs

```diff
@@ -25,75 +25,54 @@ const PERF_DFLT_INITIAL_TRACE_BUFSIZE: size_t = 1024 * 1024; // 1MiB
 thread_local! {
     /// When `Some` holds the `ThreadTraceCollector` that is collecting a trace of the current
     /// thread.
-    static THREAD_TRACE_COLLECTOR: RefCell<Option<Box<dyn ThreadTraceCollector>>> = RefCell::new(None);
+    static THREAD_TRACE_COLLECTOR: RefCell<Option<Box<dyn ThreadTracer>>> = RefCell::new(None);
```

Comment:
I prefer the specific terminology "collect" and "decode" rather than the more generic "tracer" which is kind of overloaded. Don't you agree?

Reply:
This bit ends up being changed in later commits. That said, even after this PR, the terminology we use is a bit mixed and, sometimes, confused. I'd like to address that in a future PR.

Reply:
OK. I'll see how it plays out, but I avoid "tracer" because it's so ambiguous.

The system as a whole, we often call a tracer, yet here we are talking only about the bit that collects a trace.

Reply:
Agreed: this stuff needs a rethink across the whole repo. I don't have a very firm opinion on the right thing to do other than "what we do currently can surely be improved".

Reply:
Surely undoing the rename here is what we need?

Reply:
In this case, this thread_local disappears later: but it's not just the things directly touched by this PR that have slightly confusing names. We really need to do a thorough consistency pass.